### PR TITLE
Add correlation refactor migration plugin

### PR DESCRIPTION
**Pull request by dmgaldi**
## Overview
* Migration of analyses in refactored correlation plugins.
* Resolves https://github.com/VEuPathDB/service-eda/issues/6

### CODE REVIEW

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/CorrelationRefactorPlugin.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   * @return Migrated computation.
+   */
+  private JSONObject migrateComputation(JSONObject computation) {
+    String computationType = computation.getJSONObject("descriptor").getString("type");
+    JSONObject descriptor = computation.getJSONObject("descriptor");
+    if (computationType.equals("correlationassaymetadata")) {
+      JSONObject configuration = descriptor.getJSONObject("configuration");
+      descriptor.put("type", "correlation");
+
+      configuration.put("data1", configuration.getJSONObject("collectionVariable"));
```
Comment (ryanrdoherty) [inline Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/CorrelationRefactorPlugin.java:72]: This doesn't match Danielle's description.  Should be
```
data1: {
    dataType: "collection",
    collectionSpec: <the value previously in collectionVariable>
}
```

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/CorrelationRefactorPlugin.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+      configuration.put("data2", new JSONObject());
+      configuration.getJSONObject("data2").put("dataType", "metadata");
+      return computation;
+    } else if (computationType.equals("correlationassayassay")) {
+      JSONObject configuration = descriptor.getJSONObject("configuration");
+      descriptor.put("type", "correlation");
+
+      configuration.put("data1", configuration.getJSONObject("collectionVariable1"));
+      configuration.getJSONObject("data1").put("dataType", "collection");
```
Comment (ryanrdoherty) [inline Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/CorrelationRefactorPlugin.java:84]: Same problem as above.  The collection spec needs to be nested a layer down in "collectionSpec" under "data1"

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/CorrelationRefactorPlugin.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      configuration.remove("collectionVariable1");
+
+      configuration.put("data2", configuration.getJSONObject("collectionVariable2"));
```
Comment (ryanrdoherty) [inline Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/CorrelationRefactorPlugin.java:87]: Same issue.  Needs "collectionSpec" subprop.

**Review by ryanrdoherty — CHANGES_REQUESTED** [review]

File: Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/CorrelationRefactorPlugin.java (hunk 1 above)
Comment (dmgaldi) [reply Model/src/main/java/org/gusdb/wdk/model/fix/table/edaanalysis/plugins/CorrelationRefactorPlugin.java]: Careless mistake :( thanks!

**Review by ryanrdoherty — APPROVED** [review]

**Pull request merged**